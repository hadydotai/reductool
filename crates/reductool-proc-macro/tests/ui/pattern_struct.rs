use reductool_proc_macro::aitool;

struct S {
    x: i32,
    y: i32,
}

#[aitool]
fn f(S { x, y }: S) {}

fn main() {}
