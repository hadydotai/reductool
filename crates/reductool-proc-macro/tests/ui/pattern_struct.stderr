error: unsupported parameter pattern. expected a simple identifier like `arg: T`.
       Examples of unsupported patterns: `(_: T)`, `(a, b): (T, U)`, `S { x, y }: S`.
 --> tests/ui/pattern_struct.rs:9:6
  |
9 | fn f(S { x, y }: S) {}
  |      ^^^^^^^^^^
